helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -960,10 +956,10 @@ public void process(WatchedEvent event) {
 
   private void fireAllEvents() {
     for (Entry<String, Set<IZkChildListener>> entry : _childListener.entrySet()) {
-      fireChildChangedEvents(entry.getKey(), entry.getValue());
+      fireChildChangedEvents(entry.getKey(), entry.getValue(), true);
     }
     for (Entry<String, Set<IZkDataListenerEntry>> entry : _dataListener.entrySet()) {
-      fireDataChangedEvents(entry.getKey(), entry.getValue(), OptionalLong.empty());
+      fireDataChangedEvents(entry.getKey(), entry.getValue());
```

Comment:
Another question, this is something I am not sure. No true or false passed in here as 3rd parameter in line 962.  THe function signature has not default value. What value would be then for this invocation of fireDataChangeEvents?

Reply:
This function will be invoked when session expiry. I guess it will be used to initialize the states; Thus it makes sense to assume the path exists

Reply:
I see. By default it is true.

Reply:
Actually I could remove the overloading method by passing the default value here